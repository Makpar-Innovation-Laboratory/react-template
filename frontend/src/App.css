:root {
  --primary-color: #234286 !important;
  --dark-grey: #707070 !important;
  --light-grey: #F5F5F5 !important;
  --light: rgb(223, 223, 223) !important;
  --white: #ffffff !important;
  --red: #d56969 !important;
  --green: #90b199 !important;
}

/* * {
  font-family: Arial, Helvetica, sans-serif;
} */

/* AWS Amplify authentication colors */
:root {
  --amplify-primary-color: var(--primary-color);
  --amplify-primary-tint: var(--primary-color);
  --amplify-primary-shade: var(--primary-color);
}

/* General elements styling */
.hover {
  cursor: pointer;
}
p,
h1,
h2,
h3,
h4,
h5,
h6 {
  font-weight: normal !important;
  margin: 0  !important;
}

h5 {
  padding: 3px 0 !important;
}

/* * {
  margin: 0 !important;
  padding: 0 !important;
} */
a {
  text-decoration: none !important;
  color: #0000FF !important;
}

/* Background Colors */
.background-primary {
  background: var(--primary-color) !important;
}
.background-dark {
  background: var(--dark-grey) !important;
}
.background-light {
  background: var(--light-grey) !important;
}
.bg-red {
  background: var(--red) !important;
}
.bg-green {
  background: var(--green) !important;
}

/* Text Colors */
.color-primary {
  color: #234286 !important;
}
.color-dark {
  color: #707070 !important;
}
.color-light {
  color: #bababa !important;
}
.color-white {
  color: var(--white) !important;
}
.color-red {
  color: var(--red) !important;
}
.color-green {
  color: var(--green) !important;
}

/* Header Size */
/* We
color: var(--primary-color);bsite Header - h1, Page Header - h2, Section header - h3 */
h1 {
  font-size: 30px !important;
}
h2 {
  color: var(--primary-color);
  font-size: 25px !important;
  padding: 15px 0;
}
h3 {
  color: var(--dark-grey);
  font-size: 22px !important;
}
h4 {
  color: var(--dark-grey);
  font-size: 20px !important;
}
h5 {
  color: var(--dark-grey);
  font-size: 18px !important;
}
h6 {
  color: var(--dark-grey);
  font-size: 16px !important;
}
p {
  color: var(--dark-grey);
}

/*  */
.main-wrapper {
  display: flex;
  flex-direction: column;
  text-align: center;
}

.section-wrapper {
  width: 80%; /* default */
  text-align: center;
  padding-top: 20px;
  padding-bottom: 100px;
}

/* Main-Header */
.header-container {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px 0;

  color: var(--primary-color);
}

.box {
  background-color: rgba(0, 0, 0, 0.8);
  border-radius: 10px;
  box-shadow: 0 15px 25px rgba(0, 0, 0, 0.8);
  margin: auto auto;
  padding: 40px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

/* Navigation */
.nav-container {
  margin: 0;
  width: 100vw;
  min-height: 20px;
  padding: 20px 10px !important;
}
.nav-container a{
  cursor: pointer !important;
}
.nav-items {
  width: 100%;
  min-height: 30px;
  display: flex;
  align-items: center;
  justify-content: center;
}
.nav-item {
  color: var(--white) !important;
  /* padding: 10px 20px !important; */
  margin: 10px 20px !important;
  border-bottom: 3px solid #ffffff00;
  color: var(--primary-color) !important;
}
#active {
  border-bottom: 3px solid var(--primary-color);
}
.nav-item:hover {
    /* text-decoration: underline; */
    /* background: var(--dark-grey); */
    cursor: pointer;
    border-bottom: 3px solid var(--primary-color);
}
.nav-item-white {
  color: var(--white) !important;
  margin: 10px 20px !important;
  border-bottom: 3px solid #ffffff00;
  color: var(--white) !important;
}
.nav-item-white:hover {
  cursor: pointer;
  border-bottom: 3px solid var(--white);
}
#active-white {
  border-bottom: 3px solid var(--white);
}

/* Footer Primary */

.footer{
  margin-top: 1rem;
  padding: 1rem;
  background-color: rgb(235, 195, 64);
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%;
}

/* Card Primary */
.card {
  border: 1px solid var(--light-grey);
  border-radius: 0 !important;
  min-width: 200px;
  min-height: 200px;
  width: 300px;
  background: var(--white);
}
.card-header {
  height: 25%;
  padding: 15px 0 !important;

  background: var(--primary-color) !important;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 0 !important;
  border: none;
}
.card-header p {
  color: var(--white);
}
.card-body {
  height: 75%;
  padding: 20px 15px;
}
.card-body p {
  line-height: 20px;
  color: var(--dark-grey);
}
/* Card Light */
.card-light {
  border-radius: 0;
  min-width: 200px;
  min-height: 200px;
  width: 400px;
  background: var(--white);
}
.card-header-light {
  height: 25%;
  padding: 15px 0;

  background: var(--white);
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 0 !important;
  border: none;

  border-bottom: 1px solid var(--light-grey);
}
.card-header-light p {
  font-weight: bold !important;
  color: var(--dark-grey);
}


/* Buttons */
.btn-primary {
  background: var(--primary-color) !important;
  color: #ffffff !important;
  border: 1px solid var(--primary-color) !important;
  font-size: 15px !important;
  padding: 5px 25px !important;
}
.btn-white {
  background: #ffffff !important;
  color: var(--primary-color) !important;
  border: 1px solid var(--primary-color) !important;
  font-size: 15px;
  padding: 5px 25px !important;
}
.btn-round-primary {
  background: var(--primary-color) !important;
  color: #ffffff !important;
  border: 1px solid var(--primary-color) !important;
  border-radius: 18px;
  font-size: 15px !important;
  padding: 5px 25px !important;
}
.btn-round-white {
  background: #ffffff !important;
  color: var(--primary-color) !important;
  border: 1px solid var(--primary-color) !important;
  border-radius: 18px;
  font-size: 15px !important;
  padding: 5px 25px !important;
}

/* Tables */
.table {
  width: 100%;
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}
.table h5 {
  text-align: center;
  margin: 0;
  padding: 10px 0;
}
.table-header-primary {
  display: flex;
  justify-content: center;
  padding: 10px 0;
  width: 100%;
  background: var(--primary-color);
}
.table-header-primary p {
  color: var(--white) !important;
  font-weight: 600 !important;
}
.table-header-light {
  display: flex;
  justify-content: center;
  border-top: 1px solid var(--light);
  padding: 10px 0;
  width: 100%;
}
.table-header-light p {
  font-weight: 600 !important;
}
.table-header-gray {
  display: flex;
  justify-content: center;
  padding: 6px 0;
  width: 100%;
  background: var(--light-grey) !important;
}
.row {
  display: flex;
  justify-content: center;
  border-top: 1px solid var(--light);
  padding: 6px 0 !important;
  width: 100%;
}
.row-content {
  width: 90% !important;
  display: flex;
  justify-content: space-between;
  padding: 0 !important;
}
.row:hover {
  background: #F9F9F9;
}
/* tr {
  padding: 0 13px 0 13px !important;
} */
th {
  padding: 13px 0 13px 12px !important;
}
td {
  padding: 12px 0 12px 12px !important;
}

/*  Loading Animation */
.loader {
  color: var(--primary-color);
  margin-top: 20px;
}
#load-1,
#load-2,
#load-3 {
  font-size: 100px;
  opacity: 0.5;
  animation-duration: 2s;
  animation-name: load;
  animation-iteration-count: infinite;
}
#load-2 {
  animation-delay: -0.4s;
}
#load-3 {
  animation-delay: -0.8s;
}
@keyframes load {
  0%,
  100% {
    opacity: 0.2;
  }
  50% {
    opacity: 0.5;
  }
}

/**/
input {
  padding: 5px 15px;
  border: 1px solid #BABABA;
}
input:focus, select:focus, *:focus, *:after {
  outline: none !important;
}
.btn-upload {
  background: var(--light-grey) !important;
  color: #BABABA !important;
  border: 1px solid #BABABA !important;
  font-size: 15px;
  padding: 5px 25px;
}

select {
  color: var(--dark-grey) !important;
}

#error {
  display: none;
}

form {
  background: white;
  border: 1px solid #dedede;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  margin: 0 auto;
  max-width: 500px;
  padding: 30px 50px;
}

textarea {
  font-family: var(--font-family);
  border: none;
  resize: none;
}

textarea::placeholder {
  color: var(--gray);
}

.comment-box {
  display: flex;
  flex-wrap: wrap;
  max-width: 400px;
  margin: 50px auto;
  background: var(--white);
  border-radius: var(--border-radius);
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.2);
  overflow: hidden;
  padding: 14px;
  transition: min-height var(--transition);
  max-height: 46px;
}

.comment-box.modified {
  max-height: unset;
}
.comment-field {
  color: var(--black);
  transition: transform var(--transition);
  font-size: 18px;
  width: 100%;
  outline: none;
  padding: 0;
  min-height: 60px;
  transform: translateY(-32px);
  line-height: 1;
}

.actions {
  width: 100%;
  display: flex;
  align-self: flex-end;
  justify-content: flex-end;
  margin-top: 50px;
  opacity: 0;
  transition: opacity var(--transition) 0.2s;
}

.comment-box.expanded .actions {
  opacity: 1;
}

.comment-box.expanded .header {
  transform: translateY(10px);
  opacity: 1;
  visibility: visible;
}

.comment-box.expanded .comment-field {
  transform: translateY(40px);
}

.actions button {
  margin-left: 6px;
  font-size: 14px;
  padding: 12px;
  border: none;
  line-height: 1;
}

.actions .cancel {
  background: none;
}

.post-container {
  background-color: rgb(0, 0, 0, 0.1);
  padding: 20px;
  margin: 10px;
  width: 40%;
  text-align: center;
  display: flex;
  flex-direction: column;
}
.post-content {
  min-height: 200px;
}
.post-container p {
  font-size: 1.6rem;
}
.comment-section{
  margin-top: 30px;
}
.comment-header {
  font-size: 16px;
  text-transform: uppercase;
  color: purple;
  letter-spacing: -0.3px;
  margin-bottom: 10px;
}
.comment__meta > h5 {
  font-size: 15px;
  color: purple;
  opacity: 0.7;
  margin-bottom: 3px;
  line-height: 1;
}
hr {
  background: rgba(0, 0, 0, 0.2);
  height: 3px;
}
.comment__meta > span {
  font-size: 14px;
  font-weight: bold;
  opacity: 0.5;
}
.comment__body {
  font-size: 18px;
  opacity: 0.8;
  font-family: 'Rajdhani', cursive;
}

.wrapper-class {
  padding: 1rem;
  border: 1px solid #ccc;
}
.editor-class {
  background-color:lightgray;
  padding: 1rem;
  border: 1px solid #ccc;
}
.toolbar-class {
  border: 1px solid #ccc;
}