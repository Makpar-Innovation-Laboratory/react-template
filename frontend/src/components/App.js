/* eslint-disable import/no-named-as-default */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { hot } from "react-hot-loader";

import Header from "./Header";
import ContentPanel from "./ContentPanel";

// This is a class-based component because the current
// version of hot reloading won't hot reload a stateless
// component at the top-level.

// eslint-disable-next-line react/prefer-stateless-function
class App extends Component {
  render() {
    return (
      <div className="app-container">
        <Header />
        <div id="open-right" style={{ height: "0.1rem", width: "100vw" }}></div>
        <ContentPanel />
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.element
};

export default hot(module)(App);
