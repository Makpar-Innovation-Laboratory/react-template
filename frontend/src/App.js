import React, { useState } from "react";
import "./App.css";
// import Navigation from "./Components/Navigation/Navigation";
import MaterialNavigation from './Components/Navigation/MaterialNavigation'
import Home from "./Components/Home/Home";
// import Login from "./Components/Login/Login";
import MaterialLogin from "./Components/Login/MaterialLogin"
import { Routes, Route, Navigate } from "react-router-dom";
import Auth from "./Utility/Auth";
// import SignupForm from './Components/Login/Signup';
import MaterialSignup from './Components/Login/MaterialSignup';
// import AddPost from './Components/NewsFeed/Posts/AddPost';
import MaterialAddPost from './Components/NewsFeed/Posts/MaterialAddPost';
// import Footer from './Components/Home/Footer';
import Profile from './Components/Profile/UserProfile';
// import DisplayAllPosts from './Components/NewsFeed/Posts/DisplayAllPosts'
import MaterialAllPosts from './Components/NewsFeed/Posts/MaterialAllPosts'
import DisplaySinglePost from './Components/NewsFeed/Posts/DisplaySinglePost'
import MissionStatement from './Components/Mission/MissionStatement'

export const Context = React.createContext();

export function Dashboard() {
  if (Auth.isUserAuthenticated()){ return <Home />}
  else { return <Navigate to="/Login"></Navigate> }
  return <Home />
}

/**
 * @component
 *
 * @property {}
 *
 * @description
 * description goes here
 *
 * @returns {}
 */
export default function App() {
  
  const [data, setData] = useState({results: []});
  const [isAuth, setIsAuth] = useState(false)
  const [showTable, setShowTable] = useState(false)
  const [username, setUsername] = useState("")
  const [userid, setUserid] = useState("")
  const [posts, setPosts] = useState()
  const [singlePost, setSinglePost] = useState("")
  const [markDown, setMarkdown] = useState()
  /**
   * description goes here
   * @returns {boolean}
   */
  function checkAuth() {
    if (Auth.isUserAuthenticated()) { return true; }
    else { return false; }
  }

  /**
   * description goes here
   * @param {string} input
   */
  function handleSetPosts(posts){
    setPosts(posts)
  }
  function handleRemovePosts() {
    console.log('removed')
  }
  function displaySinglePost(input) {
    setSinglePost(input)
  }
  function handleSetMarkdown(input){
    setMarkdown(input)
  }
  function updateData(inputs) {
    setData(inputs);
  }
  function updateUser(input) {
    setUsername(input)
  }
  function updateUserid(input) {
    setUserid(input)
  }
  function updateTableVis(input) {
    setShowTable(input)
  }
  /**
   * description goes here
   * @returns {}
   */

  return (
    <Context.Provider
      value={{
        
        username: username,
        updateUser: updateUser,
        userid: userid,
        updateUserid:updateUserid, 
        markDown: markDown,
        handleSetMarkdown: handleSetMarkdown,
        data,
        updateData,
        showTable,
        updateTableVis,
        isAuth: isAuth,
        setAuth: (bool) => setIsAuth(bool),
        postsState: posts,
        singlePost:singlePost,
        displaySinglePost:displaySinglePost,
        handleAddPosts: (posts) => handleSetPosts(posts),
        handleRemovePosts: () => handleRemovePosts(),
        
      }}
    >
      {/* <Router> */}
      {/* <Navigation /> */}
      <MaterialNavigation/>
      <div className="App">
        <Routes>
          <Route path='/Login' element={<MaterialLogin/>}/>
          <Route path='/Signup' element={<MaterialSignup/>}/>
          <Route path='/Mission' element={<MissionStatement/>}/>
          <Route path='/Addpost' element={<MaterialAddPost/>}/>
          <Route path='/PostArchive' element={<MaterialAllPosts/>}/>
          <Route path='/Post' element={<DisplaySinglePost/>}/>
          <Route path='/Profiles' element={<Profile/>}/>
          {/* <Route path='/' element={() => (isAuth ? <Home /> : <Redirect to="/Login" />)} />   */}
          <Route path="/" element={<Dashboard />} />
        </Routes>
      </div>
      {/* <Footer /> */}
      {/* </Router> */}
    </Context.Provider>
  );
}
