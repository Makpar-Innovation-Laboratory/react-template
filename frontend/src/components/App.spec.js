import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("JSX Tests :: <App /> Rendering", () => {

  // test("Sanity Check Snapshot Rendering", () => {

  //   const wrapper = shallow(<App />);
  //   expect(wrapper.debug()).toMatchSnapshot();
  // });
});
