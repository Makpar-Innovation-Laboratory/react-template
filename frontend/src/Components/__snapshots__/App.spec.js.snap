// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSX Tests :: <App /> Rendering Sanity Check Snapshot Rendering 1`] = `
"<div class=\\"app-container\\">
  <Header />
  <div id=\\"open-right\\" style={{...}} />
  <Connect(ContentPanel) />
</div>"
`;
